packages/nectar/src/sagas/messages/sendMessage/sendMessage.saga.test.ts

```diff
@@ -1,146 +1,89 @@
 
-  const publicChannel: PublicChannel = {
-    name: 'general',
-    description: 'description',
-    owner: 'user',
-    timestamp: 0,
-    address: 'address',
-    messagesSlice: 0
-  }
+    factory = await getFactory(store)
 
-  const communityChannels: CommunityChannels = {
-    id: communityId,
-    currentChannel: publicChannel.address,
-    channels: publicChannelsAdapter.setAll(
-      publicChannelsAdapter.getInitialState(),
-      [publicChannel]
-    ),
-    channelMessages: channelMessagesAdapter.getInitialState(),
-    unreadMessages: unreadMessagesAdapter.getInitialState()
-  }
+    community = await factory.create<
+    ReturnType<typeof communitiesActions.addNewCommunity>['payload']
+    >('Community')
 
-  test('sign and send message', async () => {
+    alice = await factory.create<ReturnType<typeof identityActions.addNewIdentity>['payload']>(
+      'Identity',
+      { id: community.id, nickname: 'alice' }
+    )
+
+    sailingChannel = (await factory.create<ReturnType<typeof publicChannelsActions.addChannel>['payload']>(
+      'PublicChannel',
+      {
+        communityId: alice.id,
+        channel: {
+          name: 'sailing',
+          description: 'Welcome to #sailing',
+          timestamp: DateTime.utc().valueOf(),
+          owner: alice.nickname,
+          address: 'sailing'
+        }
+      }
+    )).channel
+  })
+
+  test('sign and send message in current channel', async () => {
     const socket = { emit: jest.fn() } as unknown as Socket
 
+    const currentChannel = getCurrentChannel(store.getState())
+
+    const reducer = combineReducers(reducers)
     await expectSaga(
       sendMessageSaga,
       socket,
-      messagesActions.sendMessage('message')
+      messagesActions.sendMessage({ message: 'message' })
     )
-      .withReducer(
-        combineReducers({
-          [StoreKeys.Communities]: communitiesReducer,
-          [StoreKeys.Identity]: identityReducer,
-          [StoreKeys.PublicChannels]: publicChannelsReducer
-        }),
-        {
-          [StoreKeys.Communities]: {
-            ...new CommunitiesState(),
-            currentCommunity: 'id',
-            communities: communitiesAdapter.setAll(
-              communitiesAdapter.getInitialState(),
-              [community]
-            )
-          },
-          [StoreKeys.Identity]: {
-            ...new IdentityState(),
-            identities: identityAdapter.setAll(
-              identityAdapter.getInitialState(),
-              [identity]
-            )
-          },
-          [StoreKeys.PublicChannels]: {
-            ...new PublicChannelsState(),
-            channels: communityChannelsAdapter.setAll(
-              communityChannelsAdapter.getInitialState(),
-              [communityChannels]
-            )
-          }
-        }
-      )
+      .withReducer(reducer)
+      .withState(store.getState())
       .provide([
         [call.fn(parseCertificate), 'certificate'],
         [call.fn(keyFromCertificate), 'key'],
```

Comment:
Maybe change this enigmatic 'key' to 'publicKey' to keep consistency?

Reply:
In the end, it doesn't even matter. Mocked call only needs to return any value. Could be empty string in fact